aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -0,0 +1,947 @@
+
+        let nft_mint_config = borrow_global<NFTMintConfig>(@post_mint_reveal_nft);
+        let collection_config = borrow_global_mut<CollectionConfig>(@post_mint_reveal_nft);
+        let resource_signer = create_signer_with_capability(&nft_mint_config.signer_cap);
+
+        // Set up a source token if it doesn't exist yet.
+        if (!exists<SourceToken>(@post_mint_reveal_nft)) {
+            // This is the collection that contains certificate tokens, which will allow users to exchange to an NFT in the destination collection after reveal time.
+            let source_collection_name = collection_config.collection_name;
+            string::append_utf8(&mut source_collection_name, b" - source collection");
+            create_collection(&resource_signer, source_collection_name, collection_config.collection_description, utf8(b""), collection_config.collection_maximum, collection_config.collection_mutate_config);
+
+            let source_token_data_id = create_tokendata(
+                &resource_signer,
+                source_collection_name,
+                // token name
+                utf8(b""),
+                // token description
+                utf8(b""),
+                // token maximum is equal to the number of tokens in the destination collection
+                vector::length(&collection_config.token_uri),
+                // token uri
+                utf8(b""),
+                signer::address_of(&resource_signer),
+                1,
+                0,
+                create_token_mutability_config(
+                    &vector<bool>[ false, false, false, false, true ]
+                ),
+                vector<String>[string::utf8(BURNABLE_BY_OWNER)],
+                vector<vector<u8>>[bcs::to_bytes<bool>(&true)],
+                vector<String>[string::utf8(b"bool")],
+            );
+
+            move_to(&resource_signer, SourceToken { token_data_id: source_token_data_id });
+        };
+
+        // if this is the whitelist minting time
+        if (now < whitelist_mint_config.whitelist_minting_end_time) {
+            let claimer_addr = signer::address_of(nft_claimer);
+            assert!(bucket_table::contains(&whitelist_mint_config.whitelisted_address, &claimer_addr), error::permission_denied(EACCOUNT_NOT_WHITELISTED));
+            let remaining_mint_allowed = bucket_table::borrow_mut(&mut whitelist_mint_config.whitelisted_address, claimer_addr);
+            assert!(*remaining_mint_allowed >= amount, error::invalid_argument(EAMOUNT_EXCEEDS_MINTS_ALLOWED));
+            while (amount > 0) {
+                mint_source_certificate_internal(nft_claimer, whitelist_mint_config.whitelist_mint_price);
+                amount = amount - 1;
+                *remaining_mint_allowed = *remaining_mint_allowed - 1;
+            };
+        } else {
+            while (amount > 0) {
+                mint_source_certificate_internal(nft_claimer, public_mint_config.public_mint_price);
```

Comment:
there's no max on amount minted? or a global max?

Reply:
As mentioned above, there are two phases 
1. whitelist minting phase, each whitelisted address has a cap specified in whitelist config
2. public minting phase, no restriction anymore, you can mint as many as you want.  For public minting, recording all the addresses and the amount could be too high. 
We can always add new feature/ restrictions later if it turns out to be necessary.

Reply:
It does seem strange that there's no global max enforced, especially if the supply is not tracked (maximum = 0). Don't most projects have a limit on how many NFTs can be minted?

Reply:
updated yesterday to include an optional global max [here](https://github.com/aptos-labs/aptos-core/blob/34dc712d102542677063154a10bb96d8865723f8/aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move#L73)  and [here](https://github.com/aptos-labs/aptos-core/blob/34dc712d102542677063154a10bb96d8865723f8/aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move#L413)

Reply:
This is a per account limit. What if a collection has a global limit on total number of tokens but no per-account mint limit?

Reply:
+1